src/django_rich/test.py

```diff
@@ -161,18 +185,97 @@ class RichTestRunner(DiscoverRunner.test_runner):  # type: ignore [misc]
     resultclass = RichTextTestResult
 
 
+class RichTestSuite(unittest.TestSuite):
+    def _handleModuleFixture(self, test, result):
+        timing_start = time.perf_counter_ns()
+        super()._handleModuleFixture(test, result)
+        total = (time.perf_counter_ns() - timing_start) / 1e9
+        if total >= 0.005:
+            item = (total, f"{test.__module__}.setUpModule")
+            if len(result.collectedDurations) == result.slowest:
+                heapq.heappushpop(result.collectedDurations, item)
+            else:
+                heapq.heappush(result.collectedDurations, item)
+
+    def _handleClassSetUp(self, test, result):
+        timing_start = time.perf_counter_ns()
+        super()._handleClassSetUp(test, result)
+        total = (time.perf_counter_ns() - timing_start) / 1e9
+        if total >= 0.005:
+            item = (total, f"{test.__module__}.{test.__class__.__name__}.setUpClass")
+            if len(result.collectedDurations) == result.slowest:
+                heapq.heappushpop(result.collectedDurations, item)
+            else:
+                heapq.heappush(result.collectedDurations, item)
+
+
+def positive_int(value: str) -> int:
+    try:
+        num = int(value)
+    except ValueError:
+        raise argparse.ArgumentTypeError("should be an integer")
+    if num < 1:
+        raise argparse.ArgumentTypeError("should be at least 1")
+    return num
+
+
 class RichRunner(DiscoverRunner):
     pdb: bool  # django-stubs missing
 
     test_runner = RichTestRunner
+    test_suite = RichTestSuite
+
+    @classmethod
+    def add_arguments(cls, parser: argparse.ArgumentParser) -> None:
+        super().add_arguments(parser)
+        parser.add_argument(
+            "--slowest",
+            nargs="?",
+            default=20,
+            type=positive_int,
+            metavar="num_slowest",
+            help=(
+                "The number of slow tests and setup/teardown methods to "
+                + "report when --timing is used."
+            ),
+        )
+
+    def __init__(self, *args: Any, slowest: int, **kwargs: Any) -> None:
+        super().__init__(*args, **kwargs)
+        self.slowest = slowest
 
     # django-stubs bad return type, fixed in:
     # https://github.com/typeddjango/django-stubs/pull/1069
     def get_resultclass(  # type: ignore [override]
         self,
-    ) -> type[unittest.TextTestResult] | None:
+    ) -> Callable[[], unittest.TextTestResult]:
+        klass: type[RichTextTestResult]
         if self.debug_sql:
-            return RichDebugSQLTextTestResult
+            klass = RichDebugSQLTextTestResult
```

Comment:
I was struggling with how to make it work for sql and pdb. Thought it best to share what I've had sat around for a little while!